addons/stock_dropshipping/dropshipping.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+#
+#    OpenERP, Open Source Management Solution
+#    Copyright (C) 2004-2010 Tiny SPRL (<http://tiny.be>).
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp.osv import fields, osv
+from openerp.tools.translate import _
+
+class purchase_order(osv.osv):
+    _inherit = 'purchase.order'
+     
+    def action_picking_create(self, cr, uid, ids, context=None):
+        # if invoice method is 'picking' then do not allow to confirm PO
+        # what you should check is:
+        #- is the purchase and sale order linked to the same picking ?
+        #- is the sale order with invoice_method picking
+        #- is the purchase order with invoice_method picking
+        # group_id name is Sale_reference
+        # picking Origin = PO name
+        # Picking Type Id = Drop shipping
+        # Did not get picking id
+        if not context: context = {}
+        super(purchase_order, self).action_picking_create(cr, uid, ids, context=context)
+        sale_obj = self.pool.get('sale.order')
+        picking_obj = self.pool.get('stock.picking')
+        for order in self.browse(cr, uid, ids, context=context):
```

Comment:
you can check `if order.invoice_method == 'picking'` already here to avoid unnecessary work